File: docs/manual/java/guide/broker/code/docs/mb/AnotherService.java
package docs.mb;

import akka.NotUsed;

import com.lightbend.lagom.javadsl.api.*;

import static com.lightbend.lagom.javadsl.api.Service.*;

public interface AnotherService extends Service {
  ServiceCall<NotUsed, NotUsed> foo();
  @Override
  default Descriptor descriptor() {
    return named("anotherservice").withCalls(
        namedCall("/api/foo",  this::foo)
      );
  }
}

File: docs/manual/scala/guide/broker/code/docs/mb/AnotherService.scala
package docs.mb

import akka.NotUsed
import com.lightbend.lagom.scaladsl.api.{Service, ServiceCall}

/**
  *
  */
trait AnotherService extends Service {

  override final def descriptor = {
    import Service._

    named("another-service").withCalls(
      namedCall("/api/hello/:id", foo)
    ).withAutoAcl(true)

  }

  def foo: ServiceCall[NotUsed, NotUsed]

}

